pkg/deployments/tasks/20230215-single-recipient-gauge-factory-v2/test/task.fork.ts

```diff
@@ -123,168 +124,207 @@ describeForkTest('SingleRecipientGaugeFactory V2', 'mainnet', 16627100, function
-        })
-      )
-    );
+      await gaugeController.checkpoint();
+      // Gauge weight is equal to the cap, and controller weight for the gauge is greater than the cap.
+      expect(
+        await gaugeController['gauge_relative_weight(address,uint256)'](gauge.address, await currentWeekTimestamp())
+      ).to.be.gt(weightCap);
+      expect(await gauge.getCappedRelativeWeight(await currentTimestamp())).to.equal(weightCap);
+    });
 
-    // Tokens are minted for the gauge
-    const gaugeTransferEvent = expectTransferEvent(
-      await tx.wait(),
-      {
-        from: ZERO_ADDRESS,
-        to: gauge.address,
-      },
-      BAL
-    );
-    expect(gaugeTransferEvent.args.value).to.be.almostEqual(expectedEmissions);
+    it('mint & transfer tokens', async () => {
+      // The gauge has votes for this week, and it will mint the first batch of tokens. We store the current gauge
+      // relative weight, as it will change as time goes by due to vote decay.
+      const firstMintWeekTimestamp = await currentWeekTimestamp();
+
+      const calldata = gauge.interface.encodeFunctionData('checkpoint');
+
+      const zeroMintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata);
+      expectEvent.inIndirectReceipt(await zeroMintTx.wait(), gauge.interface, 'Checkpoint', {
+        periodTime: firstMintWeekTimestamp.sub(WEEK), // Process past week, which had zero votes
+        periodEmissions: 0,
+      });
+      // No token transfers are performed if the emissions are zero, but we can't test for a lack of those
+
+      await advanceTime(WEEK);
+
+      // The gauge should now mint and send all minted tokens to the Arbitrum bridge
+      const mintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata);
+      const event = expectEvent.inIndirectReceipt(await mintTx.wait(), gauge.interface, 'Checkpoint', {
+        periodTime: firstMintWeekTimestamp,
+      });
+      const actualEmissions = event.args.periodEmissions;
+
+      // The amount of tokens minted should equal the weekly emissions rate times the relative weight of the gauge
+      const weeklyRate = (await BALTokenAdmin.getInflationRate()).mul(WEEK);
+
+      // Note that instead of the weight, we use the cap (since we expect for the weight to be larger than the cap)
+      const expectedEmissions = weightCap.mul(weeklyRate).div(FP_ONE);
+      expectEqualWithError(actualEmissions, expectedEmissions, 0.001);
+
+      // Tokens are minted for the gauge
+      expectTransferEvent(
+        await mintTx.wait(),
+        {
+          from: ZERO_ADDRESS,
+          to: gauge.address,
+          value: actualEmissions,
+        },
+        BAL
+      );
+
+      // And then forwarded to the recipient
+      expectTransferEvent(
+        await mintTx.wait(),
+        {
+          from: gauge.address,
+          to: recipient,
+          value: actualEmissions,
+        },
+        BAL
```

Comment:
shouldn't you also test the last step in case mode is `FeeDistributorRecipient`?